filebeat/harvester/config.go

```diff
@@ -0,0 +1,65 @@
+package harvester
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	cfg "github.com/elastic/beats/filebeat/config"
+	"github.com/elastic/beats/filebeat/input"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+var (
+	defaultConfig = harvesterConfig{
+		BufferSize:      cfg.DefaultHarvesterBufferSize,
+		DocumentType:    cfg.DefaultDocumentType,
+		InputType:       cfg.DefaultInputType,
+		TailFiles:       cfg.DefaultTailFiles,
+		Backoff:         cfg.DefaultBackoff,
+		BackoffFactor:   cfg.DefaultBackoffFactor,
+		MaxBackoff:      cfg.DefaultMaxBackoff,
+		CloseOlder:      cfg.DefaultCloseOlder,
+		ForceCloseFiles: cfg.DefaultForceCloseFiles,
+		MaxBytes:        cfg.DefaultMaxBytes,
+	}
+)
+
+type harvesterConfig struct {
+	common.EventMetadata `config:",inline"`     // Fields and tags to add to events.
+	BufferSize           int                    `config:"harvester_buffer_size"`
+	DocumentType         string                 `config:"document_type"`
+	Encoding             string                 `config:"encoding"`
+	InputType            string                 `config:"input_type"`
+	TailFiles            bool                   `config:"tail_files"`
+	Backoff              time.Duration          `config:"backoff" validate:"min=0,nonzero"`
+	BackoffFactor        int                    `config:"backoff_factor" validate:"min=1"`
+	MaxBackoff           time.Duration          `config:"max_backoff" validate:"min=0,nonzero"`
+	CloseOlder           time.Duration          `config:"close_older"`
+	ForceCloseFiles      bool                   `config:"force_close_files"`
+	ExcludeLines         []*regexp.Regexp       `config:"exclude_lines"`
+	IncludeLines         []*regexp.Regexp       `config:"include_lines"`
+	MaxBytes             int                    `config:"max_bytes" validate:"min=0,nonzero"`
+	Multiline            *input.MultilineConfig `config:"multiline"`
+	JSON                 *input.JSONConfig      `config:"json"`
+}
```

Comment:
normally I'd prefer to have `validate`-tags to be aligned to have them stand out a little more (just add spaces in tags).